ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/BlockSimulator.java

```diff
@@ -214,14 +213,27 @@ private BlockSimulationResult processBlockStateCall(
         createBlockHashLookup(
             blockOverrides, protocolSpec, overridenBaseBlockHeader, blockHashCache);
 
+    Optional<BlockAccessListBuilder> blockAccessListBuilder =
+        protocolSpec
+            .getBlockAccessListFactory()
+            .filter(BlockAccessListFactory::isEnabled)
+            .map(BlockAccessListFactory::newBlockAccessListBuilder);
+
+    Optional<AccessLocationTracker> preExecutionAccessLocationTracker =
+        blockAccessListBuilder.map(
+            b -> BlockAccessListBuilder.createPreExecutionAccessLocationTracker());
+
     final BlockProcessingContext blockProcessingContext =
         new BlockProcessingContext(
             overridenBaseBlockHeader,
             ws,
             protocolSpec,
             blockHashLookup,
             OperationTracer.NO_TRACING);
-    protocolSpec.getPreExecutionProcessor().process(blockProcessingContext);
+
+    protocolSpec
+        .getPreExecutionProcessor()
+        .process(blockProcessingContext, preExecutionAccessLocationTracker);
```

Comment:
could the `preExecutionAccessLocationTracker` be part of the `BlockProcessingContext`?

Reply:
why it will be needed ?